src/lib/import-export/import/handlers/backup-handler-wpress.ts

```diff
@@ -88,24 +89,56 @@ async function readBlockToFile( fd: fs.promises.FileHandle, header: Header, outp
 	const outputStream = fs.createWriteStream( outputFilePath );
 
 	let totalBytesToRead = header.size;
-	while ( totalBytesToRead > 0 ) {
-		let bytesToRead = CHUNK_SIZE_TO_READ;
-		if ( bytesToRead > totalBytesToRead ) {
-			bytesToRead = totalBytesToRead;
+	let errored = false;
+	let streamEnded = false;
+
+	const errorHandler = ( err: Error ) => {
+		if ( ! errored ) {
+			Sentry.captureException( err, {
+				extra: {
+					outputFilePath,
+					header,
+					totalBytesToRead,
+				},
+			} );
+			errored = true;
 		}
+	};
 
-		if ( bytesToRead === 0 ) {
-			break;
+	const endStream = () => {
+		if ( ! streamEnded && ! outputStream.destroyed ) {
+			streamEnded = true;
+			outputStream.end();
 		}
+	};
 
-		const buffer = Buffer.alloc( bytesToRead );
-		const data = await fd.read( buffer, 0, bytesToRead );
-		outputStream.write( buffer );
+	outputStream.once( 'error', errorHandler );
 
-		totalBytesToRead -= data.bytesRead;
+	try {
+		while ( totalBytesToRead > 0 ) {
+			let bytesToRead = CHUNK_SIZE_TO_READ;
+			if ( bytesToRead > totalBytesToRead ) {
+				bytesToRead = totalBytesToRead;
+			}
+			if ( bytesToRead === 0 ) break;
+			const buffer = Buffer.alloc( bytesToRead );
+			const data = await fd.read( buffer, 0, bytesToRead );
+			if ( errored ) {
+				return;
+			}
+			if ( ! outputStream.destroyed ) {
+				outputStream.write( buffer );
+			} else {
+				return;
+			}
```

Comment:
```suggestion
			if ( outputStream.destroyed ) {
				return;
			}
			outputStream.write( buffer );
```
Nitpick: Doing an early return would improve code readability.